bottlerocket/types/src/agent_config.rs

```diff
@@ -104,6 +104,9 @@ pub struct EksClusterConfig {
     /// quoted so that it is interpreted as a JSON/YAML string (not a number).
     pub version: Option<K8sVersion>,
 
+    /// The encoded eksctl config that should be used for cluster creation.
+    pub encoded_eksctl_config: Option<String>,
```

Comment:
There's a problem here because the encoded_eksctl_config is exclusive to all the other configs we have in the struct. i.e. if I specify `cluster_name` and `encoded_eksctl_config` at the same time, then the `cluster_name` field has no effect, right? I wonder if we should make the EksClusterConfig an enum so that it can only be one way or the other:

```rust
enum EksClusterConfig {
    Structured(EksctlArguments),
    EncodedEksctlConfig(String),
}

Reply:
We still need the cluster name if we use eksctl config for looking up cluster information.

Reply:
Would it be possible to derive the cluster name from the cluster config if it's specified? Extract it out with `serde_yaml::Value` maybe? We already pull in `serde_yaml` as a dependency elsewhere in testsys.

Reply:
I think to take the idea even further, we should consider making the eksctl cluster configuration mandatory if cluster creation policy is `always` or `ifnotexists`, and the set of other fields are only used if cluster creation policy is never. 

We might need to turn `EksClusterConfig` itself into an enum, either we represent an existing cluster with `cluster_name`, `region`. Or we represent an EKS cluster that has yet to be created with an eksctl cluster config. I don't believe we really need the "create if not exist" flow (is there?). 

All of this non-blocking, but I do think it would clear things up a lot better.

Reply:
Requiring an eksctl config just makes it harder to start using the framework. Not only does a user need to learn TestSys but now they also need to learn eksctl configs.

Reply:
After thinking about it further, I think the "create if not exist" workflow is useful and we need the current interface to make that possible. 

I still strongly feel that we shouldn't have any field that, when specified, must match another value elsewhere. In this case, cluster name/region and the corresponding fields in `eksctl`'s cluster config. As much as I dislike implicit overrides, there really needs be a single source of truth that takes precedence over other fields. Take `eksctl create cluster` for example, it doesn't let you specify any flag if a configuration file is specified. Ideally we would match that behavior but it's harder to enforce through the TestSys agent configuration.

Reply:
I have updated the interface to use the eksctl config as the source of truth if it is provided. Otherwise, it will use provided fields. Now there is no duplication required. Running a few tests and then I'll push.